### ISSUE TITLE
False positive redundant expression

### ISSUE THREAD

**Issue by borkdude** [clj-kondo/clj-kondo#1183]
``` clojure
(reg-event-fx
 ::export-data-sources
 [(path db/page-id)]
 (fn [{:keys [:db]} [_]]
   (let [data (get-in db [db/analysis :term-search-count])
         _ (.log js/console data)
         data (get-in db [db/analysis :term-search-count :search/count])
         data (sort-by :count > data)
         data (map #(assoc % :label (or (get categories (:term %))
                                        (str/capitalize (:term %))))
                   data)
         csv-lines (map (fn [{:keys [:label :count]}]
                          (str (pr-str label) "," count))
                        data)
         header (str (pr-str "Data source") "," (pr-str "Count"))
         csv (str/join "\n" (cons header
                                  csv-lines))]
     (js/download csv "data-sources.csv" "text/csv")
     nil)))
```

The string `"data-sources.csv"` is a false positive

**Comment by rjerue** [comment]
I'm seeing the same error message message that is a false positive.

```
path/to/file:line#:pos#: warning: Redundant expression: :keys
path/to/file:line#:pos#: warning: Redundant expression: :some-keyword
path/to/file:line#:pos#: warning: Redundant expression: :another-keyword
```

Rolling back a kondo to `{:mvn/version "2021.02.13"}` fixes it.

I can't share the src since it's not open source and am puzzled as to why it's only happening to a few files in this massive repo. In all cases There are deconstructions in a `when-let` statement though.
